### PR TITLE
Add python3 pyside2 key

### PR DESCRIPTION
**Pull request by hellantos**
Please add the following dependency to the rosdep database.

## Package name:

python3-pyside2

## Package Upstream Source:

https://code.qt.io/cgit/pyside/pyside-setup.git/
https://code.qt.io/cgit/pyside/pyside.git/
https://code.qt.io/cgit/pyside/shiboken.git/

## Purpose of using this:

The goal of this key is to eventually replace the pyside2 specific portion of python3-qt5-bindings.
This key will install qt python libraries wrapped with PySide2. The key installs the same libraries as python3-pyqt5 see #34246. This is thought to simplify switching between pyqt and pyside2 as python wrapper providers (see https://github.com/ros-visualization/python_qt_binding/issues/114)

It will at least install wrappers for the following qt libraries:
* qtcore
* qtgui
* qthelp 
* qtnetwork
* qtprintsupport
* qtsvg
* qttest
* qtwidgets
* qtxml

In addition it will install:
* shiboken2 and libraries (generator for pyside2 wrappers)

On Fedora the pyside package is not split so this will install all libraries.

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/search?searchon=names&keywords=python3-pyside2
- Ubuntu: https://packages.ubuntu.com/
   - https://packages.ubuntu.com/search?keywords=python3-pyside2&searchon=names
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/python-pyside2/python3-pyside2/fedora-rawhide.html
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/extra/x86_64/pyside2/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/dev-python/pyside2

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
@ipa-cmh is there a reason that this PR is still in draft? Would you like initial review on that?

**Comment by ipa-cmh** [comment]
@nuclearsandwich
Its part of a bigger restrucutring around qt python bindings - see linked  issues. But I think I am ready for review now.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "ros.distro": "humble"
+}
```
Comment (jacobperron) [inline .vscode/settings.json:3]: Seems like this file was added accidentally. Please remove.

**Review by jacobperron — COMMENTED** [review]
Could you also add an entry for opensuse, as suggested by the GitHub Action.

**Comment by ipa-cmh** [comment]
Will do.

File: .vscode/settings.json (hunk 1 above)
Comment (ipa-cmh) [reply .vscode/settings.json]: Removed

**Comment by ipa-cmh** [comment]
> Could you also add an entry for opensuse, as suggested by the GitHub Action.

Opensuse seems to only distribute shiboken6 and not shiboken2. I did not find a package, that distributes the necessary shiboken binaries. So I would rather not add the key at all for opensuse.

**Review by jacobperron — APPROVED** [review]

**Review by tfoote — APPROVED** [review]

**Pull request merged**